freqtrade/persistence.py

```diff
@@ -10,52 +10,50 @@
 import arrow
 from sqlalchemy import (Boolean, Column, DateTime, Float, Integer, String,
                         create_engine)
-from sqlalchemy.engine import Engine
+from sqlalchemy import inspect
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm.scoping import scoped_session
 from sqlalchemy.orm.session import sessionmaker
 from sqlalchemy.pool import StaticPool
-from sqlalchemy import inspect
 
+from freqtrade import constants
 
 logger = logging.getLogger(__name__)
 
 _CONF = {}
 _DECL_BASE: Any = declarative_base()
 
 
-def init(config: dict, engine: Optional[Engine] = None) -> None:
+def init(config: Dict) -> None:
     """
     Initializes this module with the given config,
     registers all known command handlers
     and starts polling for message updates
     :param config: config to use
-    :param engine: database engine for sqlalchemy (Optional)
     :return: None
     """
     _CONF.update(config)
-    if not engine:
-        if _CONF.get('dry_run', False):
-            # the user wants dry run to use a DB
-            if _CONF.get('dry_run_db', False):
-                engine = create_engine('sqlite:///tradesv3.dry_run.sqlite')
-            # Otherwise dry run will store in memory
-            else:
-                engine = create_engine('sqlite://',
-                                       connect_args={'check_same_thread': False},
-                                       poolclass=StaticPool,
-                                       echo=False)
-        else:
-            engine = create_engine('sqlite:///tradesv3.sqlite')
 
+    db_url = _CONF.get('db_url', None)
+    kwargs = {}
+
+    if db_url == constants.DEFAULT_DB_DRYRUN_URL:
```

Comment:
how should we handle if `dry_run` is `True` and points to some arbitary path that differs from `DEFAULT_DB_DRYRUN_URL`

Reply:
We don't care if we are in `dry_run` at this point, this check only passes required args to `create_engine` if we have a in-memory db URL to fix threading issues with telegram rpc. I will modify this condition to check for `sqlite://` explicitly, because `DEFAULT_DB_DRYRUN_URL` is not necessary a in-memory db.